tests/rados/test_osd_replicated_inconsistency_scenario.py

```diff
@@ -0,0 +1,219 @@
+"""
+This file contains the  methods to verify the  inconsistent object functionality during scrub/deep-scrub
+in replicated pool.
+AS part of verification the script  perform the following tasks-
+   1. Creating omaps
+   2. Convert the object in to inconsistent object
+   3. Verify the functionality during scrub/deep-scrub
+"""
+
+import time
+import traceback
+
+from test_osd_ecpool_inconsistency_scenario import (
+    get_inconsistent_count,
+    set_default_param_value,
+    verify_pg_state,
+)
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
+from ceph.rados.rados_scrub import RadosScrubber
+from tests.rados.monitor_configurations import MonConfigMethods
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Test to create an inconsistent object functionality during scrub/deep-scrub in replicated pool.
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    scrub_object = RadosScrubber(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_obj)
+    pool_obj = PoolFunctions(node=cephadm)
+
+    try:
+        replicated_config = config.get("replicated_pool")
+        replicated_pool_name = replicated_config["pool_name"]
+        no_of_objects = config.get("inconsistent_obj_count")
+        pg_id = create_pool_inconsistent_object(
+            rados_obj, no_of_objects, pool_obj, **replicated_config
+        )
+
+        # Check the default values of the osd_scrub_auto_repair_num_errors and osd_scrub_auto_repair
+        auto_repair_num_value = mon_obj.get_config(
+            section="osd", param="osd_scrub_auto_repair_num_errors"
+        )
+        log.info(
+            f"Original value of osd_scrub_auto_repair_num_errors is {auto_repair_num_value}"
+        )
+        if int(auto_repair_num_value) != 5:
+            log.error(
+                "The default value of the osd_scrub_auto_repair_num_errors is not equal to 5"
+            )
+            return 1
+
+        auto_repair_value = mon_obj.get_config(
+            section="osd", param="osd_scrub_auto_repair"
+        )
+        log.info(f"Original value of osd_scrub_auto_repair is {auto_repair_value}")
+        if auto_repair_value is True:
+            log.error("The default value of the osd_scrub_auto_repair should be false")
+            return 1
+
+        # case 1: inconsistent objects are greater than the osd_scrub_auto_repair_num_errors count
+        log.info(
+            "Test scenario1: inconsistent objects greater than the osd_scrub_auto_repair_num_errors count"
+        )
+        mon_obj.set_config(
```

Comment:
Are we checking if the no of inconsistent objects on cluster is greater than 3, before setting "osd_scrub_auto_repair_num_errors" to 3? 

In the automation, We should ideally check the no of inconsistent objects on cluster, say X, and we should set the "osd_scrub_auto_repair_num_errors" param to X -1.

This was we can be sure that no repairs will occur.

Reply:
The scenario is covered in the case2.

Reply:
As explained in call, there needs to be modification based on inconsistent object count.

Reply:
Modified the code, the value of the osd_scrub_auto_repair_num_errors argument will be set based on the inconsistent object count.